### PR TITLE
Make Wordnet interoperable with various taggers and tagged corpora

### LINKED ISSUE #3392: Restrict Wordnet searches when Penn Treebank Tags are available
The NLTK Part-of-Speech tagger outputs Penn Treebank tags, which are often neglected when looking the results up in Wordnet. When a word has senses in different grammatical categories, this may lead to considering lots of spurious senses pertaining to irrelevant parts-of-speech, which could be avoided by leveraging the available _en-ptb_ tags.

But unfortunately, Wordnet searches can only be restricted by using the Wordnet parts-of-speech tags (n, v, a, r), although it would be relatively easy to provide a function that converts _en-ptb_ tags to Wordnet _pos_, by extending the conversion between en-ptb and Universal Tags (Petrov et al., 2012), which is already available through the _nltk.tag.mapping_ module.

### PR DESCRIPTION
**Pull request by ekaf**
Fix #3392 by adding a function that uses Universal Tags (Petrov et al., 2012), to convert the tags available in _nltk_data/taggers/universal_tagset_ to Wordnet Part-of-Speech. This allows to restrict Wordnet searches to the relevant grammatical category.

```
from nltk.corpus import wordnet as wn
from nltk.corpus import brown
form,tag = brown.tagged_sents()[0][17]
print((form, tag))
```
('evidence', 'NN')

Restricting the Wordnet search to _n_ discards the verbal senses:

`
print(wn.synsets(form, wn.tag2pos(tag, brown._tagset)))
`

[Synset('evidence.n.01'), Synset('evidence.n.02'), Synset('evidence.n.03')]

### CODE REVIEW

File: nltk/corpus/reader/wordnet.py [hunk 1]
```diff
@@ -2108,6 +2111,31 @@ def filter_forms(forms):
...
+            if the tag cannot be mapped to a WordNet POS.
+
+        Example:
+            >>> import nltk
+            >>> tagged = nltk.tag.pos_tag(nltk.tokenize.word_tokenize("Banks check books."))
+            >>> print([(word, tag, nltk.corpus.wordnet.tag2pos(tag)) for word, tag in tagged])
+            [('Banks', 'NNS', 'n'), ('check', 'VBP', 'v'), ('books', 'NNS', 'n'), ('.', '.', None)]
+        """
+
+        from nltk.tag import map_tag
```
[the remark this answers is not in the document]
Comment (ekaf) [reply nltk/corpus/reader/wordnet.py]: Ok, but then it is not sufficient to remove the import: it should also be added back at the top-level.

File: nltk/corpus/reader/wordnet.py [hunk 2]
```diff
@@ -2108,6 +2112,36 @@ def filter_forms(forms):
         # 2. Return all that are in the database (and check the original too)
         return filter_forms([form] + forms)
 
+    def tag2pos(self, tag, tagset="en-ptb"):
+        """
+        Convert a tag from one of the tagsets in nltk_data/taggers/universal_tagset to a
```
[the remark this answers is not in the document]
Comment (ekaf) [reply nltk/corpus/reader/wordnet.py:2117]: Maybe, but wouldn't that be too much documentation for a very simple method?

**Comment by ekaf** [comment]
Thanks Copilot for the thorough review and excellent suggestions!

**Comment by ekaf** [comment]
More from Copilot:

> Here’s an evaluation of the PR based on the provided diff:
> Summary of the Change
> 
>     Adds a tag2pos method to the WordNet corpus reader, mapping POS tags from various tagsets (like Penn Treebank and Brown) to WordNet’s POS notation (n, v, a, r), with universal tags as an intermediary.
>     Imports map_tag from nltk.tag for mapping between tagsets.
>     Defines a mapping dictionary for universal POS tags to WordNet tags.
>     Adds comprehensive docstrings and an inline usage example.
>     Includes new unit tests for both Penn Treebank and Brown tagsets to validate correct behavior, including edge cases (unmappable tags).
> 
> Strengths
> 
>     Utility: This fills a longstanding gap—WordNet often needs POS in its own notation, but users usually have Penn Treebank or Brown tags. This method makes that translation seamless.
>     Implementation: The method is concise, leverages existing NLTK functionality, and is well-documented.
>     Testing: Unit tests are thorough and cover multiple tagsets and edge cases.
>     Backwards Compatibility: This is an additive change, with no breaking API changes.
> 
> Potential Concerns
> 
>     Performance: The method is a simple wrapper, so no significant performance concern.
>     Dependency: Relies on map_tag, but this is already a part of nltk.tag and is a stable API.
>     Tagset Coverage: The docstring notes that only supported tagsets work, but that’s transparent in the documentation and usage.
> 
> Recommendation
> 
> I recommend merging this PR.
> 
>     The addition is useful, well-implemented, and safe.
>     It improves usability for a common NLP workflow without risk to existing functionality.
>     Tests ensure reliability.
> 
> Optional suggestion: If you want to further improve it, you could add a direct reference or link to the NLTK documentation for map_tag in the docstring, but this is not required for merge.
> 
> In summary:
> This is a well-targeted, user-friendly improvement. Merging it will benefit users who need to map POS tags to WordNet’s format in their NLP pipelines.

**Comment by ekaf** [comment]
Even though Copilot sounds superficially credible, this still has a serious bug, and even fails in a nasty way:

```
from nltk.corpus import wordnet as wn
wn.tag2pos('NOUN','universal')

```

> LookupError: 
> **********************************************************************
>   Resource universal_tagset not found.
>   Please use the NLTK Downloader to obtain the resource:
> 
>   >>> import nltk
>   >>> nltk.download('universal_tagset')
>   
>   For more information see: https://www.nltk.org/data.html
> 

The download error is a diffent bug, and was often seen in other contexts. But we need to support 'universal' as input tag format.

**Comment by ekaf** [comment]
LGTM

**Comment by stevenbird** [comment]
Great contribution, thanks @ekaf

**Comment by ekaf** [comment]
And thanks Copilot for a very useful review!
